api-metastore/src/main/java/org/zalando/nakadi/service/SchemaEvolutionService.java

```diff
@@ -99,6 +107,48 @@ private void collectErrorMessages(final ValidationException e,
     public EventType evolve(final EventType original, final EventTypeBase eventType) throws SchemaEvolutionException {
         checkEvolutionIncompatibilities(original, eventType);
 
+        switch (eventType.getSchema().getType()) {
+            case JSON_SCHEMA:
+                return evolveJsonSchema(original, eventType);
+            case AVRO_SCHEMA:
+                return evolveAvroSchema(original, eventType);
+            default:
+                throw new RuntimeException("unsupported type for evolving " + eventType.getSchema().getType());
+        }
+    }
+
+    private EventType evolveAvroSchema(final EventType original, final EventTypeBase eventType) {
```

Comment:
should indicate throwing of `SchemaEvolutionException`

Reply:
done